### PR TITLE
Quota alerting

### PR DESCRIPTION
**Pull request by ghost**
Ensuring that missing utilization metrics won't trigger critical errors in the context of the quota feature, unless quotas are actually in place. Currently, a fresh install (without any metrics) will have quota enabled, but not used; yet, we see the alert: 

image

This fix relies on the count of bucket/account with quota. This count must then be > 0 for the alert to be triggered. This metric is computed by the s3 report instance periodically.

### CODE REVIEW

File: monitoring/alerts.yaml [hunk 1]
```diff
@@ -141,7 +144,9 @@ groups:
   - alert: QuotaMetricsNotAvailable
     expr: |
       avg(avg_over_time(s3_cloudserver_quota_utilization_service_available{namespace="${namespace}",service="${service}"}[1m]))
-        < ${quotaUnavailabilityThreshold}
+        < ${quotaUnavailabilityThreshold} and
```
Comment (francoisferrand) [inline monitoring/alerts.yaml:147]: why would this not be true on a properly running system (i.e. in which conditions)?

File: monitoring/alerts.yaml [hunk 2]
```diff
@@ -141,7 +144,9 @@ groups:
   - alert: QuotaMetricsNotAvailable
     expr: |
       avg(avg_over_time(s3_cloudserver_quota_utilization_service_available{namespace="${namespace}",service="${service}"}[1m]))
```
Comment (francoisferrand) [inline monitoring/alerts.yaml]: what is the point of averaging over the last minute?
we scrap resources every 30 second, so does not add much... and alerts have their own mecanism to smooth over time (i.e. the `for` parameter)

File: monitoring/alerts.yaml (hunk 1 above)
Comment (ghost) [reply monitoring/alerts.yaml:147]: I'm not sure I understand this comment, but I'm guessing it's related to our discussion earlier: so I am closing this comment, please feel free to reopen if needed

File: monitoring/alerts.yaml (hunk 2 above)
Comment (ghost) [reply monitoring/alerts.yaml]: Indeed, I:
- Removed the avg_over_time over the last minute
- Removed the `for` for the warning version of this alert

File: .github/workflows/alerts.yaml [hunk 3]
```diff
@@ -28,5 +28,5 @@ jobs:
           alert_file_path: monitoring/alerts.yaml
           test_file_path: ${{ matrix.tests.file }}
           alert_inputs: >-
-            namespace=zenko,service=artesca-data-connector-s3api-metrics,replicas=3
+            namespace=zenko,service=artesca-data-connector-s3api-metrics,reportJob=artesca-data-ops-report-handler,replicas=3
```
Comment (francoisferrand) [inline .github/workflows/alerts.yaml]: should be put on multiple lines, to improve readability (c.f. https://github.com/scality/backbeat/blob/development/8.6/.github/workflows/alerts.yaml#L18-L34)

```suggestion
          alert_inputs: |
            namespace=zenko
            service=artesca-data-connector-s3api-metrics
            reportJob=artesca-data-ops-report-handler
            replicas=3
```

(this requires updted `scality/action-prom-render-test@1.0.3`, but this should be already the case on tip of dev branches: did you rebase on latest changes?)

File: monitoring/alerts.test.yaml [hunk 4]
```diff
@@ -229,6 +231,105 @@ tests:
...
         exp_alerts: []
 
+  # QuotaMetricsNotAvailable (case with account quota)
+  ##################################################################################################
+  - name: Quota metrics not available (account quota)
+    interval: 1m
+    input_series:
+      - series: s3_cloudserver_quota_utilization_service_available{namespace="zenko",service="artesca-data-connector-s3api-metrics"}
+        values: 1+1x6 0+0x20 1+1x6
+      - series: s3_cloudserver_quota_accounts_count{namespace="zenko",job="artesca-data-ops-report-handler"}
```
Comment (francoisferrand) [inline monitoring/alerts.test.yaml:241]: should also add test when `s3_cloudserver_quota_accounts_count` is not present (i.e. maybe add a few `undefined` at the beginning of the series)

File: .github/workflows/alerts.yaml (hunk 3 above)
Comment (ghost) [reply .github/workflows/alerts.yaml]: yes I am up to date, so I will bump the version also

File: monitoring/alerts.yaml [hunk 5]
```diff
@@ -140,9 +143,10 @@ groups:
   # but not available for at least half of the S3 services during the last minute
   - alert: QuotaMetricsNotAvailable
     expr: |
-      avg(avg_over_time(s3_cloudserver_quota_utilization_service_available{namespace="${namespace}",service="${service}"}[1m]))
-        < ${quotaUnavailabilityThreshold}
-    for: 1m
+      avg(s3_cloudserver_quota_utilization_service_available{namespace="${namespace}",service="${service}"})
+        < ${quotaUnavailabilityThreshold} and
+        (max(s3_cloudserver_quota_buckets_count{namespace="${namespace}", job=~"${reportJob}"}) > 0 or
+        max(s3_cloudserver_quota_accounts_count{namespace="${namespace}", job=~"${reportJob}"}) > 0)
```
Comment (francoisferrand) [inline monitoring/alerts.yaml]:
```suggestion
         max(s3_cloudserver_quota_accounts_count{namespace="${namespace}", job=~"${reportJob}"}) > 0)
```

Comment (francoisferrand) [inline monitoring/alerts.yaml]:
```suggestion
        (max(s3_cloudserver_quota_buckets_count{namespace="${namespace}", job="${reportJob}"}) > 0 or
        max(s3_cloudserver_quota_accounts_count{namespace="${namespace}", job="${reportJob}"}) > 0)
```

Comment (francoisferrand) [reply monitoring/alerts.yaml:149]: should we filter on `service` instead? this is usually what we do, though job and service are most often the same...

File: monitoring/alerts.yaml [hunk 6]
```diff
@@ -153,8 +157,10 @@ groups:
   # but not available during the last 10 minutes
   - alert: QuotaMetricsNotAvailable
     expr: |
-      avg(avg_over_time(s3_cloudserver_quota_utilization_service_available{namespace="${namespace}",service="${service}"}[1m]))
-        < ${quotaUnavailabilityThreshold}
+      avg(s3_cloudserver_quota_utilization_service_available{namespace="${namespace}",service="${service}"})
+        < ${quotaUnavailabilityThreshold} and
+        (max(s3_cloudserver_quota_buckets_count{namespace="${namespace}", job=~"${reportJob}"}) > 0 or
+        max(s3_cloudserver_quota_accounts_count{namespace="${namespace}", job=~"${reportJob}"}) > 0)
```
Comment (francoisferrand) [inline monitoring/alerts.yaml]:
```suggestion
        (max(s3_cloudserver_quota_buckets_count{namespace="${namespace}", job="${reportJob}"}) > 0 or
         max(s3_cloudserver_quota_accounts_count{namespace="${namespace}", job="${reportJob}"}) > 0)
```

**Review by francoisferrand — APPROVED** [review]
LGTM, just a few cleanups

File: .github/workflows/alerts.yaml (hunk 3 above)
Comment (ghost) [reply .github/workflows/alerts.yaml]: Done and passing, I bumped the action version and applied your suggestions

Comment (francoisferrand) [reply .github/workflows/alerts.yaml]: I must have missed it when making the ghcr migration...

can you also bump `checkout@v4` ?

Comment (ghost) [reply .github/workflows/alerts.yaml]: Sure, let me add a commit

Comment (ghost) [reply .github/workflows/alerts.yaml]: Done

**Review by welansari — APPROVED** [review]

**Review by benzekrimaha — APPROVED** [review]

**Pull request merged**